apps/imgviewer.c

```diff
@@ -100,143 +116,174 @@ void resize_finish(int w, int h) {
 	yutani_flip(yctx, window);
 }
 
+int file_exists(char * filename){
+	return access(filename, F_OK);
+}
 
-int main(int argc, char * argv[]) {
+int is_readable(char * filename){	
 
-	static struct option long_opts[] = {
-		{"help",   no_argument,       0, '?'},
-		{0,0,0,0}
-	};
-
-	if (argc > 1) {
-		/* Read some arguments */
-		int index, c;
-		while ((c = getopt_long(argc, argv, "h", long_opts, &index)) != -1) {
-			if (!c) {
-				if (long_opts[index].flag == 0) {
-					c = long_opts[index].val;
-				}
-			}
-			switch (c) {
-				case 'h':
-					usage(argv);
-					exit(0);
-					break;
-				default:
-					break;
-			}
-		}
-	}
-
-	yctx = yutani_init();
-	if (!yctx) {
-		fprintf(stderr, "%s: failed to connect to compositor\n", argv[0]);
+	if ( file_exists( filename) != 0 ){
+		fprintf(stderr, "%s: not exist! error_info: %s\n", filename, strerror( errno ) );
 		return 1;
 	}
-	init_decorations();
+	
+	FILE * tmp = fopen( filename, "r" );
 
-	struct decor_bounds bounds;
-	decor_get_bounds(NULL, &bounds);
-
-	decor_left_width = bounds.left_width;
-	decor_top_height = bounds.top_height;
-	decor_right_width = bounds.right_width;
-	decor_bottom_height = bounds.bottom_height;
-	decor_width = bounds.width;
-	decor_height = bounds.height;
-
-	if (strstr(argv[optind],".jpg")) {
-		load_sprite_jpg(&img, argv[optind]);
-	} else {
-		load_sprite(&img, argv[optind]);
-	}
-	if (!img.width) {
-		fprintf(stderr, "%s: failed to open image %s\n", argv[0], argv[optind]);
+	if ( tmp == NULL ) {
+        fprintf(stderr, "%s: failed to open! error_info %s\n", filename, strerror( errno ) );		
 		return 1;
-	}
-	img.alpha = ALPHA_EMBEDDED;
+    } 
+	
+	fclose( tmp );	
+	return  0;
+}
 
-	width = img.width;
-	height = img.height;
+int main(int argc, char * argv[]) {
+	int opt;
+	while ((opt = getopt(argc, argv, "?:-:")) != -1) {
+		switch (opt) {
+			case '-':
+				if (!strcmp(optarg,"version")) {				
```

Comment:
Again, this seems to be something you took from bim. Bim does not follow the standard ToaruOS coding style.

Additionally, you've discarded the `--help` argument, changing behavior in something unrelated to what you were addressing in this change.